lisa/node.py

```diff
@@ -160,24 +154,49 @@ def check_sudo_password_required(self) -> None:
                 # So check twice to get two kinds of prompt
                 password_prompts = []
                 for i in range(1, 3):
-                    process = self._execute(
-                        f"echo {constants.LISA_TEST_FOR_SUDO}",
-                        shell=True,
-                        sudo=True,
-                        no_info_log=True,
-                    )
-                    result = process.wait_result(10)
+                    result = self._execute_sudo_test()
```

Comment:
It needs a refactoring on `support_sudo`. The `check_sudo_password_required` should be moved to remote node. The remote node overrides `support_sudo`, if the node.support_sudo return false. It will be clearer, and it doesn't need many logic on cast remote node.

```python
# Node class
    def support_sudo(self) -> bool:
        self.initialize()

        # check if sudo supported
        if self._support_sudo is None:
            if self.is_posix:
                process = self._execute("command -v sudo", shell=True, no_info_log=True)
                result = process.wait_result(10)
                if result.exit_code == 0:
                    self._support_sudo = True
                else:
                    self._support_sudo = False
                    self.log.info("node doesn't support sudo, may cause failure later.")
            else:
                # set Windows to true to ignore sudo asks.
                self._support_sudo = True

        return self._support_sudo

# Remote node class
    def support_sudo(self) -> bool:
        if self._support_sudo is None:
                result = super.support_sudo
                if not result:
                    self.check_sudo_password_required()